src/main/java/testapp/core/User.java

```diff
@@ -1,30 +1,40 @@
 package testapp.core;
 
-public class User {
-    private int id;
+import com.google.appengine.api.datastore.Entity;
+
+import java.io.Serializable;
+
+public class User implements Serializable {
+    private long ucn;
     private String name;
```

Comment:
yes, but you make the distinction that ucn is not another system's id, which it is, and I don't see why do we have to restrict our system to use more then one id, and on top of that if we choose to use another systems id, we have to validate if the provided id belongs to the outer system, so at least we would have to validate that id if it follows the rules of the outer system, for me this makes thinks too complicated.